### DISCUSSION TITLE
cannot match `tea.KeySpace`

### DISCUSSION THREAD

**Discussion by tgirod** [charmbracelet/bubbletea#288] [Q&A] [answered]
I think (correct me if I'm wrong) it is not possible to match any event to `tea.KeySpace`.

**Comment by meowgorithm** [comment, marked as answer]
Hi! You actually just match on a literal space:

```go
// It's usually simplest to just use the Stringer interface
switch msg.String() {
case " ":
    // It's a space!
}

// The super verbose way
if msg.Type == tea.KeyRunes && string(msg.Runes) == " " {
   // It's a space!
}
```

**Comment by tgirod** [comment]
Thanks, I figured that part by myself! ;)

What I meant is, pressing [space] in bubbletea will produce a `Key{Type: KeyRunes}` event, but no keystroke will ever produce a `Key{Type: KeySpace}`, and I find it misleading.

In my first program I wanted [space] to trigger a special behaviour, so naturally I went for something like this:

```
switch msg.Type {
case tea.KeySpace:
    // triggers something special
case tea.KeyRunes:
    // append to the model's state
}
```

... and it took me some time to realize that this won't match the space key.

**Reply by meowgorithm** [reply]
Ah! I see what you mean now. `KeySpace` is residue from early Bubble Tea development. @bashbunni is sorting this out #289 right now.

**Comment by bashbunni** [comment]
Hey, just an update that these changes were merged to `main` earlier today, so you can use `tea.KeySpace` or `tea.KeySpace.String()` now. Thanks for letting us know!

**Comment by bashbunni** [comment]
Hi @tgirod we got rid of `KeySpace` as it was causing other issues with our Bubbles components. I'd recommend using @meowgorithm 's original suggestion of using a `" "` to check for a space.

**Reply by meowgorithm** [reply]
Technically speaking `KeySpace` is still in the codebase but it will be removed in the next release.

**Reply by tgirod** [reply]
ok, thanks for the heads-up! In the end I've simplified stuff, and not using it anymore.
